remix-astwalker/src/sourceMappings.ts

```diff
@@ -1,28 +1,60 @@
 import { isAstNode, AstWalker } from './astWalker';
-import { AstNode, Location } from "./types";
+import { AstNode, LineColPosition, LineColRange, Location } from "./types";
+const util = require("remix-lib").util;
 
 export declare interface SourceMappings {
   new(): SourceMappings;
 }
 
 /**
- * Break out fields of an AST's "src" attribute string (s:l:f)
- * into its "start", "length", and "file index" components.
+ * Turn an character offset into a LineColPosition
+ *
+ * @param {Number} offset  - the character offset to convert.
+ * @returns {LineColPosition}
+ */
+export function lineColPositionFromOffset(offset: number, lineBreaks: Array<number>): LineColPosition {
+  let line: number = util.findLowerBound(offset, lineBreaks);
```

Comment:
we are basically moving this from https://github.com/ethereum/remix/blob/528ad905205efd409bf165e8cd7b5b11a0017a58/remix-lib/src/sourceMappingDecoder.js#L92 , which seems needed. But we can't keep both code, we should deprecate the one in `remix-lib`

Reply:
I am assuming this is just a comment. If there was some action I should take here, let me know.

Reply:
I think once `remix-astWalker` gets a bit mature, we can start removing existing codes we are copying/migrating from.

Reply:
Would love to see that and will help to make sure this is integrated to your atom plugin. See also #1224, #1225, #1226, #1227 which @yann300  asked me to log as issues. 

Also although I haven't added the whole walker function could probably be `asyc` which might improve performancs. I haven't logged that as an issue though. 

#1230 is not really related but it is a big annoyance in vscode because it brings in keccak which vscode barfs on badly. (I need to isolate what's up there and report to Microsoft. I recall it has been reported in the past, but not isolated.)

Reply:
Also I should say that I am sure there are lots of opportunities for making things efficient. But first we should have the basic functionality working, and right now in the plain `walk` routine I don't think that is the case. (This is a little bit an artifact of the current solc AST design which I believe will be documented better in the future.)

Reply:
> I think once remix-astWalker gets a bit mature, we can start removing existing codes we are copying/migrating from.

sounds good